go-controller/pkg/ovn/egressgw.go

```diff
@@ -572,6 +573,10 @@ func (oc *DefaultNetworkController) deletePodSNAT(nodeName string, extIPs, podIP
 
 	node, err := oc.watchFactory.NodeCoreInformer().Lister().Get(nodeName)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
if the node isnt found in the informer, dont we still have to cleanup ovn dbs?

Reply:
good question. I will look at the code, but if node does not exist at this point, 1) sync would be the one in charge of deleting the ovn dbs and 2) even if the nats where left behind, it would be referred to a logical router that is not local to this db.

Reply:
I double checked, @martinkennelly . Indeed this is a logic that only exists in for supporting ovn-ic.
And if node does not exist, the db is not going to exist either on 4.14, or at the worst case will be benign.

Reply:
If the node is deleted, then our node handler should delete the GR which will remove the SNATs as well.